### PR TITLE
Discover jenkins with autoip host name instead of host:port

### PR DESCRIPTION
**Pull request by takirala**
Address [COPS-3395](https://jira.mesosphere.com/browse/COPS-3395)

### CODE REVIEW

File: scripts/bootstrap.py [hunk 1]
```diff
@@ -30,7 +30,8 @@ def populate_jenkins_config_xml(config_xml, master, name, host, port, role, user
 
     _find_and_set(mesos, './master', master)
     _find_and_set(mesos, './frameworkName', name)
-    _find_and_set(mesos, './jenkinsURL', "http://{}:{}".format(host, port))
+    # This used to be host and port. Switching over to DNS Name to address COPS-3395.
+    _find_and_set(mesos, './jenkinsURL', "http://{}.marathon.autoip.dcos.thisdcos.directory".format(name))
```
Comment (colin-msphere) [inline scripts/bootstrap.py]: Can the format string (`"http://{}.marathon.autoip.dcos.thisdcos.directory"`) be a constant at the top of the file? It is used again on line `121`.

File: scripts/bootstrap.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import xml.etree.ElementTree as ET
 
 
-def populate_jenkins_config_xml(config_xml, master, name, host, port, role, user):
+def populate_jenkins_config_xml(config_xml, master, name, role, user):
```
Comment (colin-msphere) [inline scripts/bootstrap.py]: The function documentation needs updated with these changes.

File: scripts/bootstrap.py (hunk 1 above)
Comment (colin-msphere) [inline scripts/bootstrap.py]: The Universe config for Jenkins defines two ports and this used to specifically use the nginx port. How is the DNS address selecting the correct port?

File: scripts/bootstrap.py [hunk 3]
```diff
@@ -158,7 +159,7 @@ def _get_xml_root(config_xml):
     return tuple([tree, root])
 
 
-def _find_and_set(element, term, new_text):
+def _find_and_set(element, term, new_text, write_if_empty=False):
```
[the remark this answers is not in the document]
Comment (takirala) [reply scripts/bootstrap.py]: Done.

**Comment by colin-msphere** [comment]
What versions of DC/OS support the `autoip` DNS?

**Comment by takirala** [comment]
@colin-msphere I think 1.8 and above. 
[1.8](https://docs.mesosphere.com/1.8/usage/service-discovery/dns-overview/)
[1.9](https://docs.mesosphere.com/1.9/networking/dns-overview/)
[1.10](https://docs.mesosphere.com/1.10/networking/dns-overview/)
[1.11](https://docs.mesosphere.com/1.11/networking/DNS/)
cc @jeid64

**Pull request merged**